scapy/contrib/modbus.py

```diff
@@ -773,10 +773,10 @@ class ModbusADURequest(Packet):
                    XByteField("unitId", 0xff)]  # 0xFF (recommended as non-significant value) or 0x00
 
     def guess_payload_class(self, payload):
-        function_code = int(payload[0].encode("hex"), 16)
+        function_code = int(bytes_hex(chb(payload[0])), 16)
```

Comment:
WTF (I'm talking about the existing code)? Looks like `function_code = orb(payload[0])` is enough, or did I miss something?

Reply:
You're very right